robocop_ng/helpers/ryujinx_log_analyser.py

```diff
@@ -210,15 +210,20 @@ def __get_hardware_info(self):
 
                 case "ram":
                     ram_match = re.search(
-                        r"RAM:\s(Total)?\s([^;\n\r]*)\s;\s(Available)?\s(\d+)",
+                        r"RAM: Total ([\d.]+) (MiB|GB) ; Available ([\d.]+) (MiB|GB)",
                         self._log_text,
                         re.MULTILINE,
                     )
-                    if ram_match is not None and ram_match.group(2) is not None:
-                        self._hardware_info[setting] = ram_match.group(2).rstrip()
-                        ram_available = ram_match.group(4).rstrip()
-                        if ram_available.isnumeric():
+                    if ram_match is not None:
+                        try:
+                            ram_available = float(ram_match.group(3))
+                            if ram_match.group(4) == "GB":
+                                ram_available *= 1024
+
+                            self._hardware_info[setting] = f"{ram_match.group(1)} {ram_match.group(2)}"
                             self._ram_available_mib = int(ram_available)
+                        except ValueError:
+                            pass  # ram_match.group(3) couldn't be parsed as a float. Ignore the error and carry on.
```

Comment:
```suggestion
                        except ValueError:
                            # ram_match.group(3) couldn't be parsed as a float.
                            self._hardware_info[setting] = "Error"
                            self._ram_available_mib = -1
```
In case this happens, could you set these values, so we are able to see that there was an issue?
Otherwise the setting would be displayed as "Unknown" which could mean the log didn't contain the information as well.

Reply:
Done!